src/main/java/model/LottoRank.java

```diff
@@ -0,0 +1,39 @@
+package model;
+
+import java.util.Arrays;
+
+public enum LottoRank {
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    FAILED(0, 0);
+
+    private final int winningNumberCount;
+    private final long prizeMoney;
+
+    LottoRank(int winningNumberCount, long prizeMoney) {
+        this.winningNumberCount = winningNumberCount;
+        this.prizeMoney = prizeMoney;
+    }
+
+    public static LottoRank getRank(int winningNumberCount, boolean containsBonusBall) {
+        if (winningNumberCount == THIRD.winningNumberCount && !containsBonusBall) {
+            return THIRD;
+        }
+
+        return Arrays.stream(values())
+                .filter(lottoRank -> lottoRank.winningNumberCount == winningNumberCount)
+                .findFirst()
+                .orElseGet(() -> FAILED);
```

Comment:
```suggestion
                .orElse(FAILED);
```
Optional의 orElse()와 orElseGet()의 차이를 학습해보시면 좋을 것 같아요. (이미 아실 것 같기도 한데)
상수값이라면 orElse()를 써도 괜찮습니다 :)

Reply:
이 경우에는 반환되는 값이 메소드가 아닌 FAILED라는 상수 값이므로 orElse()를 써도 된다는 말씀이신가요?

Reply:
네네 맞습니다. orElse()에 메서드가 있다면 무조건 실행되어서 원하지 않는 인스턴스가 생성될 수도 있거든요 :)